### PR TITLE
Refactor: use flattened 'interaction' object only in react component

### PR DESCRIPTION
**Pull request by hlibbabii**
#### Proposed Changes

* Do not flatten-unflatten interactions objects (which is inside the experience object) back and forth. Unflatten it once when creating an experience in the react component and always use the unflattened structure after that.

Slack discussion:
https://trustroots.slack.com/archives/C0A3Q15SS/p1608857982047000?thread_ts=1608764438.035300&cid=C0A3Q15SS

#### Testing Instructions

* Check that the experiences tab in someone's profile loads without errors and that interaction labels 'met', 'hostedMe', 'hostedThem' are present. Check everything is fine with the labels also when creating an experience.

### CODE REVIEW

File: modules/references/client/components/read-references/Meta.js [hunk 1]
```diff
@@ -36,8 +36,6 @@ export default function Meta({ met, hostedMe, hostedThem, recommend }) {
 }
 
 Meta.propTypes = {
-  hostedMe: PropTypes.bool.isRequired,
-  hostedThem: PropTypes.bool.isRequired,
-  met: PropTypes.bool.isRequired,
+  interactions: PropTypes.object.isRequired,
```
Comment (simison) [inline modules/references/client/components/read-references/Meta.js:39]: Could do something like this to describe the shape here:

https://github.com/Trustroots/trustroots/blob/07249bba5adfe650c9c62469617f8440cb137eeb/modules/users/client/users.prop-types.js#L1-L7

Totally also ok if not.

Comment (hlibbabii) [reply modules/references/client/components/read-references/Meta.js:39]: oh, absolutely, also for experience object then, and experience pairs. I guess we can do it in a separate PR

**Review by simison — APPROVED** [review]
Looks good and tests well 👍

File: modules/references/client/components/read-references/Meta.js (hunk 1 above)
Comment (simison) [reply modules/references/client/components/read-references/Meta.js:39]: Yes, totes optional and fine in follow-ups. 👍

**Pull request merged**